MediaRequestHandler.load sometimes fails

Results in a null pointer exception.

```
java.lang.RuntimeException: An error occured while executing doInBackground()
at android.os.AsyncTask$3.done(AsyncTask.java:300)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
at java.util.concurrent.FutureTask.run(FutureTask.java:242)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:841)
Caused by: java.lang.NullPointerException: bitmap == null
at com.squareup.picasso.Utils.checkNotNull(Utils.java:123)
at com.squareup.picasso.RequestHandler$Result.<init>(RequestHandler.java:58)
at de.danoeh.antennapod.core.asynctask.PicassoProvider$MediaRequestHandler.load$71fa0c91(PicassoProvider.java:167)
at com.squareup.picasso.BitmapHunter.hunt(BitmapHunter.java:206)
at de.danoeh.antennapod.core.service.playback.PlaybackService$3.doInBackground$10299ca(PlaybackService.java:798)
at de.danoeh.antennapod.core.service.playback.PlaybackService$3.doInBackground(PlaybackService.java:787)
at android.os.AsyncTask$2.call(AsyncTask.java:288)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
... 3 more
```

The code in question is:
```
            Bitmap bitmap = null;
            MediaMetadataRetriever mmr = null;
            try {
                mmr = new MediaMetadataRetriever();
                mmr.setDataSource(data.uri.getPath());
                byte[] image = mmr.getEmbeddedPicture();
                if (image != null) {
                    bitmap = decodeStreamFromByteArray(data, image);
                }
            } catch (RuntimeException e) {
                Log.e(TAG, "Failed to decode image in media file", e);
            } finally {
                if (mmr != null) {
                    mmr.release();
                }
            }

            if (bitmap == null) {
                Log.wtf(TAG, "THIS SHOULD NEVER EVER HAPPEN!!");
            }
            return new Result(bitmap, Picasso.LoadedFrom.DISK);
```

Basically, we all the way to the end, past the if (bitmap == null) check and, the bitmap is null, so there's an exception.

Not sure why this is happening.  We could potentially return some static bitmap for cases like this...

--- a/core/src/main/java/de/danoeh/antennapod/core/asynctask/PicassoProvider.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/asynctask/PicassoProvider.java
@@ -162,6 +162,14 @@ public class PicassoProvider {
             }
 
             if (bitmap == null) {
+                // this should never, happen, but sometimes it does, so fallback
+                // check for fallback Uri
+                String fallbackParam = data.uri.getQueryParameter(PicassoImageResource.PARAM_FALLBACK);
+                if (fallbackParam != null) {
+                    Uri fallback = Uri.parse(fallbackParam);
+                    bitmap = decodeStreamFromFile(data, fallback);
+                }
+
                 Log.wtf(TAG, "THIS SHOULD NEVER EVER HAPPEN!!");
             }
             return new Result(bitmap, Picasso.LoadedFrom.DISK);

--- a/core/src/main/java/de/danoeh/antennapod/core/asynctask/PicassoProvider.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/asynctask/PicassoProvider.java
@@ -169,8 +169,6 @@ public class PicassoProvider {
                     Uri fallback = Uri.parse(fallbackParam);
                     bitmap = decodeStreamFromFile(data, fallback);
                 }
-
-                Log.wtf(TAG, "THIS SHOULD NEVER EVER HAPPEN!!");
             }
             return new Result(bitmap, Picasso.LoadedFrom.DISK);
 

--- a/core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java
@@ -414,6 +414,14 @@ public class FeedMedia extends FeedFile implements Playable {
         if (hasEmbeddedPicture()) {
             Uri.Builder builder = new Uri.Builder();
             builder.scheme(SCHEME_MEDIA).encodedPath(getLocalMediaUrl());
+
+            if (item != null && item.getFeed() != null) {
+                final Uri feedImgUri = item.getFeed().getImageUri();
+                if (feedImgUri != null) {
+                    builder.appendQueryParameter(PARAM_FALLBACK, feedImgUri.toString());
+                }
+            }
+
             return builder.build();
         } else {
             return item.getImageUri();

